pkgs/tools/system/rocm-smi/default.nix

```
@@ -1,30 +1,52 @@
-{ lib, buildPythonApplication, fetchFromGitHub }:
+{ lib, stdenv, fetchFromGitHub, cmake, python3 }:
 
-buildPythonApplication rec {
+stdenv.mkDerivation rec {
   pname = "rocm-smi";
-  version = "4.0.0";
+  version = "4.1.0";
 
   src = fetchFromGitHub {
     owner = "RadeonOpenCompute";
-    repo = "ROC-smi";
+    repo = "rocm_smi_lib";
     rev = "rocm-${version}";
-    hash = "sha256-0QqaBMkqRVEl89x3hvWQGAgt7LbtMZPhuf7KenQYHaQ=";
+    hash = "sha256-LEaC1XhmyoVWrpL05MhgN02LVT2rLKdnw9g2QdfM/uE=";
   };
 
-  format = "other";
+  nativeBuildInputs = [ cmake python3.pkgs.wrapPython ];
 
-  dontConfigure = true;
-  dontBuild = true;
+  postPatch = ''
+    # Upstream ROCm is installed in an /opt directory. For this reason,
+    # it does not completely follow FHS layout, creating top-level
+    # rocm_smi, oam, and bindings top-level directories. Since rocm-smi
+    # is a package that is typically installed, we change the paths to
+    # follow FHS more closely.
 
```

```suggestion
  # Upstream ROCm is installed in an /opt directory. For this reason,
  # it does not completely follow FHS layout, creating top-level
  # rocm_smi, oam, and bindings top-level directories. Since rocm-smi
  # is a package that is typically installed, we change the paths to
  # follow FHS more closely.
  postPatch = ''
```

This reduces the final derivation size a bit.

---

But than half of the comments are in the indented block and the others outside?

---

If they are written in the phase the are treated as part of the command and updating them causes a rebuild because they are included in the drv. This also grows the size of the drv file which should be avoided especially for bigger comments. If they comments are just a line or two this is not a big problem but for bigger ones this can be avoided.

---

I don't think the size of the drv is much of an issue, given that the whole derivation is 3KiB. Rebuilds when changing comments is annoying, but these comments don't really get updated anyway, unless the situations changes and then other parts of the derivation will change as well.

Of course, if a phase has a single comment, I would agree that it is better to put it outside the phase itself. But if they have multiple comments that are part of a 'discourse', I think it is nicer to have them in the same scope.